### ISSUE TITLE
[Wave Collect] [Remove Free Workspace Creation] Add tryNewDot.classicRedirect NVP to All Accounts Invited to Workspaces Created from NewDot

### ISSUE THREAD

**Issue by tgolen** [Expensify/App#37150]
Coming from the design doc section [here](https://docs.google.com/document/d/16ptT3EbBMHCP7TzUBlRADWuBASo3pL12d_Z0wYr38c4/edit#bookmark=id.ayrrkwzgfgsy)

Update inviteUsers [here](https://github.com/Expensify/Web-Expensify/blob/b046e32ef13e6a4a7156140e3fc4894cb0c6a118/lib/PolicyAPI.php#L5680) to loop through all invited people and check:

- Was the policy created from NewDot?
- Has the employee not received the classicRedirect NVP before?
- If both are true, then set the classicRedirect NVP

Psuedo code:
```
        if ($policy->isPolicyExpenseChatEnabled() && $isFromNewDot) {
            foreach ($employeeList as $employee) {
                $userAuthToken = AuthTokenManager::getEmailToken($employee->getEmail());
                $nvp = NVPStore::get($authToken, NVP::TRY_NEWDOT) ?? [];
                if (!isset($nvp['classicRedirect'])) {
                    $nvp['classicRedirect'] = [
                        'timestamp' => (new DateTime())->format(SQL_DATETIME),
                    ];
                    NVPStore::set($userAuthToken, NVP::TRY_NEWDOT, $nvp);
                }
            }
        }
```

- Add an integration test that verifies if someone is invited to a policy from NewDot that the classicRedirect NVP is added to their account.

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0166b750a8a56d48e9</li>
        <li>Upwork Job ID: 1761070445154676736</li>
        <li>Last Price Increase: 2024-02-23</li>
    </ul>
</details>

**Comment by Julesssss** [comment]
Hey @tgolen, I started on this today and just wanted to clarify a couple of things before I begin as I'm not sure the changes from the doc will be sufficient for all cases.

**How invites work today for a NewDot workspace:**
A) New user, invited from NewDot -- ✅ NVP added
B) New user, invited from OldDot -- ❌ NVP not added
C) Existing user, invited from NewDot -- ❌ NVP not added
D) Existing user, invited from OldDot -- ❌ NVP not added

After adding the [changes from the doc](https://docs.google.com/document/d/16ptT3EbBMHCP7TzUBlRADWuBASo3pL12d_Z0wYr38c4/edit#bookmark=id.ayrrkwzgfgsy), the NVP will be added in case **C**. But the NVP will still not be set for cases **B & D**.

Is that expected? Or should I look for a solution for that case, in addition to the changes from the doc? I think I would just need to locate the policy [here](https://github.com/Expensify/Web-Expensify/blob/615af2562a56adbe4556f6902c9d5abafc4b751f/api.php#L1550-L1576), and set the redirect NVP if the workspace was created in NewDot.

**Comment by tgolen** [comment]
Hey Jules,

I think we only need to cover the `A` and `C` case. I forgot to look at the `InviteUsers` command when I wrote the doc. The place you linked to in the code will be a good place to add that NVP.

Thanks!

**Comment by Julesssss** [comment]
WIP PR: https://github.com/Expensify/Web-Expensify/pull/41028

This is working for most cases, I just need to handle a couple of edge cases and ensure the test cases will catch regressions:

1) New user, invited from NewDot
2) Existing user, invited from NewDot
3) Existing user who dismissed redirect modal, invited from newdot

Also, we may need to set the NVP `dismissed` to false when setting the NVP for users with `dismissed: false` -- I just need to look into how the NVP is used elsewhere.

**Comment by Julesssss** [comment]
PR was updated today. 

- I need to find a workaround to a test that [compares](https://github.com/Expensify/Web-Expensify/pull/41028#discussion_r1506028639) a timestamp set during the test
    - @tgolen do you think mocking or simply hardcoding the classicRedirect timestamp value be the best course of action?
- An unrelated test is failing: `  [alrt] Bedrock\Client - Failed to connect or send the request; not retrying because we are out of retries ~~ host: 'localhost' message: 'Could not connect to Bedrock host localhost:8008. Error: 111 Connection refused'`

**Comment by Julesssss** [comment]
No progress here today, a previously held Hybrid app task took priority. I'll finish this off tomorrow and submit for review.

**Comment by Julesssss** [comment]
I'm now blocked on adding QA test steps for users that have been redirected to NewDot via the Wave 5 TryNewDot modal, seeking help [here](https://expensify.slack.com/archives/C05DWUDHVK7/p1709293794322459). Also I'm still getting unrelated test errors after multiple runs...

**Comment by Julesssss** [comment]
I figured out test steps for the existing user flow. Just holding on the flakey test now.

In the meantime I'll help out with some Wave 5 hybrid build testing in the meantime

**Comment by Julesssss** [comment]
This is on HOLD as my more critical Build/test Hybrid app issue and general Hybrid app testing is no longer blocked. I believe the critical [issue](https://github.com/Expensify/Expensify/issues/371824) to be a more important use of my time, but I can set aside a few hours tomorrow to work on the CI test failure tomorrow to try and finish this off.

**Comment by trjExpensify** [comment]
@Julesssss can we close this?

**Comment by trjExpensify** [comment]
Bump, @Julesssss. Confirming this can be closed? I only see a Web PR linked to this issue.

**Comment by Julesssss** [comment]
Yep, I missed your last comment somehow
